### ISSUE TITLE
Label and Annotation key validation does not allow the "/" char, resulting in label with truncated key

### ISSUE THREAD

**Issue by fgiudici** [rancher/dashboard#12720]
**Setup**
- Rancher version: v2.9.3
- Rancher UI Extensions: 
- Browser type & version: Mozilla Firefox 132.0.1

**Describe the bug**
In the Registration creation component of the Elemental UI extension it is not possible to add Label and Annotation templates with keys having the "/" character.
In particular as soon as the "/" char is typed in the `key` field, the row disappears from the UI.
Moreover, if the resource is created, the resulting resource (looking at the yaml) will contain the templates label/annotation with the key (which will be anyway truncated with what was typed in till the UI row disappeared).
See attached video.

**To Reproduce**

Add a Label/Annotation with a key containing "/"  (e.g., `elemental.cattle.io/InternalIP`).
**Result**
Row disappear while typing, but label is present with a truncated label key in the created resource

**Expected Result**

The Label and Annotation rows do not disappear from the UI when keys are valid.
The labels key validation should match the Kubernetes label keys syntax: see https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set .

**Screenshots**
Video:
https://github.com/user-attachments/assets/cbe1cb93-b748-4ff8-aef2-7e577ce87f62

**Additional context**

After checking with @aalves08, he suggested to track here the issue as the component used in the Elemental UI belongs here.

**Comment by richard-cox** [comment]
I've noticed this on other use cases of the label and annotation component, but it's when focus is removed from the key input that the row is removed. Looking at the yaml the key is added (with empty string)

**Comment by rak-phillip** [comment]
I don't think that this is directly related to #11226, but it's something to look out for; we previously modified input on every keypress in the yaml editor and we could be doing something similar here.

**Comment by richard-cox** [comment]
Linking #13070

**Comment by gaktive** [comment]
https://github.com/rancher/dashboard/issues/10958 is also in the mix here as we see more validation bugs come in around this area.

**Comment by gaktive** [comment]
Of note, PR https://github.com/rancher/dashboard/pull/13071 has come up through @skanakal to help with https://github.com/rancher/dashboard/pull/13070 though there may be something to leverage here. Still, as noted across multiple tickets, this area has a few bugs around it.

**Comment by richard-cox** [comment]
Linking e2e tests that's probably blocked on this https://github.com/rancher/dashboard/issues/13191

**Comment by richard-cox** [comment]
Note - this is NOT a customer issue. the JIRA was created for other tracking purposes (?)

**Comment by richard-cox** [comment]
Updated title to be inline with JIRA.

The core issue of special characters behaving badly could not be reproduced in 2.11.0 (dev of), so marking as fixed in this version. Whoever picks this up to test should just validate that part

The issue of handling system labels and annotations has been spun out to https://github.com/rancher/dashboard/issues/13655

**Comment by yonasberhe23** [comment]
Reopening: Able to reproduce this on head image (Rancher v2.12-62362dfd905a75f77865a47aff683a195cc6ac66-head / master `6f23a10`)

https://github.com/user-attachments/assets/c9665a9d-d86d-4c0b-b57c-2394e13231b9

**Comment by richard-cox** [comment]
There's a mix up with the general issue description and the specific example

Below is the current state (with caveats, see https://github.com/rancher/dashboard/issues/13655)
- special characters like `/` CAN be used in label keys
- label keys that are classed as system ones (which coincidentally mostly contain `/`) CANNOT be used a
  -  This has been spun out to https://github.com/rancher/dashboard/issues/13655) 

For this issue we need to validate just `/` outside of system ones (see #13655 for details)

**Comment by yonasberhe23** [comment]
Thanks for the clarification @richard-cox.

---
Tested in Rancher v2.12-62362dfd905a75f77865a47aff683a195cc6ac66-head / master `6f23a10`

| Label Key | Expected Result | Result 
| --- | --- | --- |
| `mycustom/label` | Accepted  | Pass
| `example.com/env` | Accepted  | Pass
| `cattle.io/name` | Restricted | Pass
